# Data from the APHYLO paper

The dataset `candidate_trees.rds` was extracted directly from
this repository [USCbiostats/aphylo-simulations](https://github.com/USCbiostats/aphylo-simulations),
and corresponds to the list of trees proteins that were used in the paper
"On the automatic annotation of gene functions using phylogenetic trees".

The file `mcmc_partially_annotated_no_prior.rds` corresponds to the MCMC parameter
estimates using experimental evidence with a uniform prior for the parameters.

# Data from SIFTER

The folders `deaminase` and `sulfotransferase` were downloaded directly from
the sup mats from the 2011 SIFTER paper.

The `nudix` family annotations were transcribed directly from the paper, in
particular, the tree featured in page number X.

`gene_association.goa_uniprot.80.txt` which is formated in GAF 1.0
(see http://www-legacy.geneontology.org/GO.format.gaf-1_0.shtml)

Was downloaded from ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/old/UNIPROT/ version 80

The file hundredfamilies was downloaded from supplemental materials of the 2011
paper published in Genome Research. 

https://genome.cshlp.org/content/21/11/1969/suppl/DC1

https://ndownloader.figstatic.com/files/6649398 (see also https://doi.org/10.1371/journal.pcbi.0010045.sd001)
From the paper website https://doi.org/10.1371/journal.pcbi.0010045 (Original 2015 paper from plos comp bio)

The nudix family tree, under the `hundred` folder, was downloaded from PFAM
version 20 (as mentioned in the paper)
ftp://ftp.ebi.ac.uk/pub/databases/Pfam/releases/Pfam20.0/.

